apps/remote_control/lib/lexical/remote_control/build/document.ex

```diff
@@ -1,11 +1,36 @@
 defmodule Lexical.RemoteControl.Build.Document do
   alias Lexical.Document
+  alias Lexical.RemoteControl.Build
   alias Lexical.RemoteControl.Build.Document.Compilers
 
   @compilers [Compilers.Config, Compilers.Elixir, Compilers.EEx, Compilers.HEEx, Compilers.NoOp]
 
   def compile(%Document{} = document) do
     compiler = Enum.find(@compilers, & &1.recognizes?(document))
-    compiler.compile(document)
+    me = self()
+
+    {pid, ref} =
+      spawn_monitor(fn ->
+        result = compiler.compile(document)
+        send(me, {:result, result})
+      end)
+
+    receive do
+      {:result, result} ->
+        flush_normal_down(ref, pid)
+        result
+
+      {:DOWN, ^ref, :process, ^pid, {exception, stack}} ->
+        diagnostic = Build.Error.error_to_diagnostic(document, exception, stack, nil)
+        diagnostics = Build.Error.refine_diagnostics([diagnostic])
+        {:error, diagnostics}
+    end
+  end
+
+  defp flush_normal_down(ref, pid) do
+    receive do
+      {:DOWN, ^ref, :process, ^pid, :normal} ->
+        :ok
+    end
```

Comment:
We should add an `after` block with a small timeout (50-100ms?) that throws a runtime error if the normal down isn't received. (I think it should always be received, but if it's not, this will block forever.)

Reply:
I went back and forth on this. On one hand, I appreciate somewhat defensive coding, but on the other, if we don't receive a message, a large part of the set of guarantees the erlang runtime system provides are false.

Reply:
Note, we moved to `Process.demonitor(ref, [:flush])`.